shared/src/clients/mail-client.class/index.ts

```diff
@@ -51,11 +48,11 @@ export default class MailClient {
         },
         ...(option?.extraSmtpHeaders || {}), // guard against undefined extraSmtpHeaders value
       } as { [key: string]: any }
-      if (input.referenceId !== undefined) {
```

Comment:
`referenceId` felt like another potentially confusing name for a variable that holds the message ID. I felt it was cleaner to simply use the term `messageId` now that this term will always refer to the table primary key value and not the SES ID -- every channel (i.e. SMS, Telegram) seem to use the term `messageId` to refer to the canonical, unique identifier for a message, so this feels consist.

Reply:
**[change request, subjective]** given our latest discussion — should we use a different name for the table primary key id?
- the word `messageId` is quite strongly associated with AWS SES's ID (see [here](https://docs.aws.amazon.com/ses/latest/dg/send-email-concepts-process.html)) and I think is the source of confusion for when I was tracing through the codebase on my own
- we will be logging the AWS SES's ID as part of the logging (and it's returned by the `sendEmail` method), so we'll have both variable names in the codebase

I think sth verbose would be the most clear, e.g. `emailMessagesId` or `emailTransactionalMessagesId`

Reply:
- After this commit the name `messageId` -- within the context of email -- should only ever refer to the table primary key. The `sendEmail` message now returns a success/failure boolean rather than the SES ID because no callsites were using the SES ID.
- The other channels (e.g. SMS, Telegram) also appear to use the term `messageId` to refer to the primary identifier for a message.
- On the basis of the above two points I think it's okay to stick with `messageId`? For the one case where we want to log the SES ID I think we can use an appropriately qualified variable name (e.g. `sesMessageId`) but it will only ever be used in the local function scope (e.g. `sendEmail`), so I don't think we're at risk of it causing confusion with the `messageId` variable.

Reply:
as spoken, I still prefer referring to the primary key id directly because in all the other instances, the `messageId` always refers to the ID given by our downstream provider, so it's easy to get confused! nonetheless, am ok to sticking with this naming as long as it's commented properly in our code :)